tests/test_preprints.py

```diff
@@ -2348,6 +2356,44 @@ def test_auth_download(self):
         observed_url.port = expected_url.port
         assert_equal(expected_url, observed_url)
 
+    @mock.patch('addons.base.views.PreprintDownload.record_for_preprint')
+    def test_metrics_download(self, mock_metrics):
+        user = UserFactory()
+        session = Session(data={'auth_user_id': user._id})
+        session.save()
+        cookie = itsdangerous.Signer(settings.SECRET_KEY).sign(session._id)
+
+        url = self.build_url(cookie=cookie)
+        res = self.app.get(url, auth=Auth(user=user))
+        assert res.status_code == 200
+        mock_metrics.assert_called_once_with(origin='Foles',
+                                             path=self.preprint.primary_file.path,
+                                             preprint=self.preprint,
+                                             referrer='Ertz',
+                                             uri='Sproles',
+                                             user=user,
+                                             user_agent='Agholor',
+                                             version=u'1')
```

Comment:
Good tests - asserting that PreprintDownload.record_for_preprint is being called correctly. 

Also what I did to verify locally: set up my index template based on `develop`, without these new fields. Then I downloaded some preprints. I checked out your branch, then ran `sync_metrics` to update the index template.  Then I downloaded some more preprints.  Verified that newly downloaded preprint documents are sending the new fields.